### PR TITLE
Make amp-img srcset select based on screen width.

### PR DESCRIPTION
**Pull request by cramforce**
This aligns `amp-img` with regular `srcset` on `img` and makes src selection independent of layout. Eventually this would allow server side rendering the `amp-img` element.

The change tends to download larger images than the old algorithm. For a typical full-width is does not make a difference, though.

This change may need to be experiment guarded if we decide to go for it.

CC @honeybadgerdontcare

### CODE REVIEW

**Comment by dvoytenko** [comment]
@cramforce Is this the case though? In some testing I did originally, the browsers have disagreed in how srcset was selected.

**Comment by cramforce** [comment]
@dvoytenko Even better. Could we decide that this is the correct way? :)

**Comment by robinpokorny** [comment]
@cramforce I think the selection should be based on `sizes` attribute, which in HTML5 must be present when `srcset` with width descriptors is present, cf. https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-sizes.

**Comment by cramforce** [comment]
@robinpokorny It certainly should be, but we need to remain backward compatible.

**Comment by robinpokorny** [comment]
@cramforce I see, then I agree that it should be window width 👍

File: builtins/amp-img.js [hunk 1]
```diff
@@ -174,7 +174,7 @@ export class AmpImg extends BaseElement {
     if (this.getLayoutWidth() <= 0) {
       return Promise.resolve();
     }
-    const src = this.srcset_.select(this.getLayoutWidth(), this.getDpr()).url;
+    const src = this.srcset_.select(this.win.screen.width, this.getDpr()).url;
```
Comment (dvoytenko) [inline builtins/amp-img.js]: Ok, but it should be window width, same as with `@media (min-width)`.

**Comment by dvoytenko** [comment]
Though I can confirm that all browsers support W-based srcset w/o sizes.

File: builtins/amp-img.js (hunk 1 above)
Comment (cramforce) [inline builtins/amp-img.js]: Done. I kept the fallback, but maybe that was overzealous.

**Comment by cramforce** [comment]
Dima: Could you review?

**Pull request merged**